gflownet.py

```diff
@@ -487,25 +490,92 @@ def __init__(
     def parameters(self):
         return self.model.parameters()
 
-    def backward_sample(self, env, policy="model", done=False):
+    def forward_sample(self, envs, times, policy="model", model=None, temperature=1.0):
+        """
+        Performs a forward action on each environment of a list.
+
+        Args
+        ----
+        env : list of GFlowNetEnv or derived
+            A list of instances of the environment
+
+        times : dict
+            Dictionary to store times
+
+        policy : string
+            - model: uses self.model to obtain the sampling probabilities.
+            - uniform: samples uniformly from the action space.
+
+        model : torch model
+            Model to use as policy if policy="model"
+
+        temperature : float
+            Temperature to adjust the logits by logits /= temperature
+        """
+        if not isinstance(envs, list):
+            envs = [envs]
+        states = [env.state2obs() for env in envs]
+        mask_invalid_actions = [env.get_mask_invalid_actions() for env in envs]
+        random_action = self.rng.uniform()
```

Comment:
random_action is not used later, is it ?

Reply:
True, good catch! I forgot to delete that line. Will do it.